app/src/main/java/com/github/se/eventradar/ui/component/ReusableComponents.kt

```diff
@@ -144,162 +146,170 @@ fun EventCard(event: Event, onCardClick: (String) -> Unit) {
 
         // Slider for free selection
         Row(
             verticalAlignment = Alignment.CenterVertically,
-            horizontalArrangement = Arrangement.Start) {
+            horizontalArrangement = Arrangement.Start,
+            modifier = Modifier.testTag("freeSwitchRow")) {
               Text(
-                  text = "Free:  ",
+                  text = stringResource(id = R.string.free_events_label),
                   style =
                       TextStyle(
                           fontSize = 16.sp,
-                      ))
+                      ),
+                  modifier = Modifier.testTag("freeSwitchLabel"))
               Switch(
-                  checked = uiState.freeEventsFilter,
-                  onCheckedChange = { uiState.freeEventsFilter = it },
-              )
+                  checked = uiState.isFreeSwitchOn,
+                  onCheckedChange = { onFreeSwitchChanged() },
+                  modifier = Modifier.testTag("freeSwitch"))
             }
 
-        Spacer(modifier = Modifier.height(4.dp))
-
         // Buttons for category selection
         Text(
-            text = "Category: ",
-            modifier = Modifier.weight(1f),
+            text = stringResource(id = R.string.category_label),
+            modifier = Modifier.weight(1f).testTag("categoryLabel"),
             style =
                 TextStyle(
                     fontSize = 16.sp,
                 ))
 
         Row(
             verticalAlignment = Alignment.CenterVertically,
-            horizontalArrangement = Arrangement.Start) {
-              CategorySelection()
-            }
+            horizontalArrangement = Arrangement.Start,
+            modifier = Modifier.testTag("categoryRow"),
+        ) {
+          CategorySelection(
+              uiState = uiState,
+              modifier = Modifier.testTag("categoryOptionsColumn"),
+          )
+        }
 
         Spacer(modifier = Modifier.height(4.dp))
 
         // Button to apply filter
-        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.Center) {
-          Button(
-              onClick = {
-                // Apply filter
-                onRadiusChange(uiState.radiusInputFilter)
-                onFreeSelectionChange(uiState.freeEventsFilter)
-                onCategorySelectionChange(uiState.categorySelectionFilter)
-                onCategorySelectionChange(uiState.categorySelectionFilter)
-              }) {
-                Text("Apply")
-              }
-        }
+        Row(
+            modifier = Modifier.fillMaxWidth().testTag("filterApplyRow"),
+            horizontalArrangement = Arrangement.Center) {
+              Button(
+                  onClick = { onFilterApply() }, modifier = Modifier.testTag("filterApplyButton")) {
+                    Text(stringResource(id = R.string.filter_apply))
+                  }
+            }
       }
     }
   }
 }
 
+// TO DO: Need to convert category selection to MVVM and test
```

Comment:
This method is not tested. If you don't plan on testing it in the future, you can remove the comment.